src/VkDispatchTable.h

```diff
@@ -0,0 +1,4617 @@
+/* 
+ * Copyright � 2021 Cody Goodson (contact@vibimanx.com)
+ * 
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+ * documentation files (the �Software�), to deal in the Software without restriction, including without
+ * limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies
+ * of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+ * 
+ * The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+ * 
+ * THE SOFTWARE IS PROVIDED �AS IS�, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT
+ * LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+ * WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ * 
+ */
+
+// This file is a part of VkBootstrap
+// https://github.com/charles-lunarg/vk-bootstrap
+
+
+#pragma once
+
+#include <vulkan/vulkan.h>
+
+namespace vkb {
+
+struct DispatchTable {
+	DispatchTable() = default;
+	DispatchTable(VkDevice const& device, PFN_vkGetDeviceProcAddr const& procAddr) : device(device) {
```

Comment:
VkDevice is an opaque handle, eg a uint64_t. You don't need to pass by `const&`, pass by value is fine. Same goes for the proc addr, its is similarly small.